contracts/TokenManager.sol

```diff
@@ -189,6 +273,262 @@ contract TokenManager is System, IApplication {
+      else if (idx == 4) mirrorSynPackage.bep20Supply   = iter.next().toUint();
+      else if (idx == 5) mirrorSynPackage.bep20Decimals = uint8(iter.next().toUint());
+      else if (idx == 6) mirrorSynPackage.mirrorFee     = iter.next().toUint();
+      else if (idx == 7) {
+        mirrorSynPackage.expireTime = uint64(iter.next().toUint());
+        success = true;
+      }
+      else break;
+      idx++;
+    }
+    return (mirrorSynPackage, success);
+  }
+
+  function decodeMirrorAckPackage(bytes memory msgBytes) internal pure returns(MirrorAckPackage memory, bool) {
+    MirrorAckPackage memory mirrorAckPackage;
+    RLPDecode.Iterator memory iter = msgBytes.toRLPItem().iterator();
+    bool success = false;
+    uint256 idx=0;
+    while (iter.hasNext()) {
+      if (idx == 0)      mirrorAckPackage.mirrorSender   = iter.next().toAddress();
+      else if (idx == 1) mirrorAckPackage.bep20Addr      = iter.next().toAddress();
+      else if (idx == 2) mirrorAckPackage.bep20Decimals  = uint8(iter.next().toUint());
+      else if (idx == 3) mirrorAckPackage.bep2Symbol     = bytes32(iter.next().toUint());
+      else if (idx == 4) mirrorAckPackage.mirrorFee      = iter.next().toUint();
+      else if (idx == 5) {
+        mirrorAckPackage.errorCode  = uint8(iter.next().toUint());
+        success = true;
+      }
+      else break;
+      idx++;
+    }
+    return (mirrorAckPackage, success);
+  }
+
+  function mirror(address bep20Addr, uint64 expireTime) payable public returns (bool) {
+    require(ITokenHub(TOKEN_HUB_ADDR).getBep2SymbolByContractAddr(bep20Addr) == bytes32(0x00), "already bound");
+    require(!mirrorPendingRecord[bep20Addr], "mirror pending");
+    uint256 miniRelayFee = ITokenHub(TOKEN_HUB_ADDR).getMiniRelayFee();
+    require(msg.value%TEN_DECIMALS == 0 && msg.value>=mirrorFee.add(miniRelayFee), "msg.value must be N * 1e10 and greater than sum of miniRelayFee and mirrorFee");
+    require(expireTime>=block.timestamp + 120 && expireTime <= block.timestamp + 86400, "expireTime must be two minutes later and one day earlier");
+    uint8 decimals = IBEP20(bep20Addr).decimals();
+    uint256 totalSupply = IBEP20(bep20Addr).totalSupply();
+    require(convertToBep2Amount(totalSupply, decimals) <= MAX_BEP2_TOTAL_SUPPLY, "too large total supply");
+    string memory name = IBEP20(bep20Addr).name();
+    bytes memory nameBytes = bytes(name);
+    require(nameBytes.length>=1 && nameBytes.length<=32, "name length must be in [1,32]");
+    string memory symbol = IBEP20(bep20Addr).symbol();
+    bytes memory symbolBytes = bytes(symbol);
+    require(symbolBytes.length>=MINIMUM_BEP20_SYMBOL_LEN && symbolBytes.length<=MAXIMUM_BEP20_SYMBOL_LEN, "symbol length must be in [3,8]");
+    for (uint8 i = 0; i < symbolBytes.length; i++) {
+      require((symbolBytes[i]>='a' && symbolBytes[i]<='a') || (symbolBytes[i]>='A' && symbolBytes[i]<='Z'), "symbol must not contain non-alphabet");
```

Comment:
The condition of symbol on BC is `token symbol should be alphanumeric`. So number is ok, can we allow number?

Reply:
BC require all symbol is alphanumeric only

Reply:
But the implementation here: `(symbolBytes[i]>='a' && symbolBytes[i]<='a')` is confusing and number is invalid.

Reply:
My mistake, it should be `(symbolBytes[i]>='a' && symbolBytes[i]<='z')`